src/solver/ts-generator/availability.cpp

```diff
@@ -584,48 +547,114 @@ void GeneratorTempData::operator()(Data::Area& area, Data::ThermalCluster& clust
 
     if (derated)
         cluster.series.timeSeries.averageTimeseries();
-
-    if (archive)
-        writeResultsToDisk(area, cluster);
-
-    cluster.calculationOfSpinning();
 }
 } // namespace
 
-std::vector<Data::ThermalCluster*> getAllClustersToGen(Data::AreaList& areas,
+std::vector<Data::ThermalCluster*> getAllClustersToGen(const Data::AreaList& areas,
                                                        bool globalThermalTSgeneration)
 {
     std::vector<Data::ThermalCluster*> clusters;
 
-    areas.each([&clusters, &globalThermalTSgeneration](Data::Area& area) {
-        for (auto& cluster : area.thermal.list.all())
+    areas.each([&clusters, &globalThermalTSgeneration](const Data::Area& area) {
+        for (const auto& cluster : area.thermal.list.all())
             if (cluster->doWeGenerateTS(globalThermalTSgeneration))
                 clusters.push_back(cluster.get());
     });
 
     return clusters;
 }
 
-bool GenerateThermalTimeSeries(Data::Study& study,
-                               std::vector<Data::ThermalCluster*> clusters,
-                               uint year,
-                               Solver::IResultWriter& writer)
+listOfLinks getAllLinksToGen(Data::AreaList& areas)
+{
+    listOfLinks links;
+
+    areas.each([&links](const Data::Area& area) {
+        std::ranges::for_each(area.links, [&links](auto& l) {
+            links.emplace_back(l.second, linkDirection::direct);
+            links.emplace_back(l.second, linkDirection::indirect);
+        });
+    });
+
+    return links;
+}
+
+void writeResultsToDisk(const Data::Study& study,
+                        Solver::IResultWriter& writer,
+                        const Matrix<>& series,
+                        const std::string& savePath)
+{
+    if (study.parameters.noOutput)
+        return;
+
+    std::string buffer;
+    series.saveToBuffer(buffer, 0);
+    writer.addEntryFromBuffer(savePath, buffer);
+}
+
+bool generateThermalTimeSeries(Data::Study& study,
+                               const std::vector<Data::ThermalCluster*>& clusters,
+                               Solver::IResultWriter& writer,
+                               const std::string& savePath)
 {
     logs.info();
     logs.info() << "Generating the thermal time-series";
-    Solver::Progression::Task progression(study, year, Solver::Progression::sectTSGThermal);
 
-    auto* generator = new GeneratorTempData(study, progression, writer);
+    bool archive = study.parameters.timeSeriesToArchive & Data::timeSeriesThermal;
 
-    generator->currentYear = year;
+    auto generator = GeneratorTempData(study, study.parameters.nbTimeSeriesThermal);
 
     // TODO VP: parallel
     for (auto* cluster : clusters)
-        (*generator)(*cluster->parentArea, *cluster);
+    {
+        AvailabilityTSGeneratorData clusterInterface(cluster);
```

Comment:
`AvailabilityTSGeneratorData clusterInterface(cluster);` 

to be replaced with : 

```C++
AvailabilityTSGeneratorData tsConfigData(cluster);
generator.generateTS(*cluster->parentArea, tsConfigData);
```

Same thing with **linkInterface**

Reply:
Same thing with **linkInterface**